elastic/kibana
# [Index management] Project level retention support

Fixes: https://github.com/elastic/kibana/issues/193160

## Summary

This PR adds support for project level retention in index management data streams UI. At the moment **this enforcement is only** in place for stateful security projects.

##### How to test
* Start ES with `node scripts/es serverless --tag latest --projectType security -E data_streams.lifecycle.retention.factory_default=200d -E data_streams.lifecycle.retention.factory_max=200d -E data_streams.lifecycle.retention.max=200d` and kibana with `yarn serverless-security`
* Navigate to console and create a data stream following [this instructions](https://www.elastic.co/guide/en/elasticsearch/reference/current/set-up-a-data-stream.html#create-component-templates)
* Navigate to `Index Management` -> `Data streams`
* Verify that:
  * Changes are only applied to security solution
  * A callout is shown above the data streams page with a link to cloud (you might need to configure cloud url locally in order to have it working) 
  * The callout is dismissable and the setting for it is stored in localstorage.
  * When editing the data retention, there should be an enforced limit in the UI based on the global retention returned by the lifecycle api.
  
  
```
# Configure cloud locally by adding the following to config/kibana.dev.yml and restarting kibana
xpack.cloud.id: 'test'
xpack.cloud.base_url: "https://cloud.elastic.co"
xpack.cloud.deployment_url: "/deployments/deploymentId"
```


<details>
<summary>Screenshots</summary>

![Screenshot 2024-09-24 at 16 05 49](https://github.com/user-attachments/assets/b44e6b24-0f8f-4a41-b55a-5f79fae70f48)
![Screenshot 2024-09-24 at 16 05 54](https://github.com/user-attachments/assets/240b6c53-16d0-48d2-a86c-58c7e2e16b4d)
![Screenshot 2024-09-30 at 15 09 49](https://github.com/user-attachments/assets/b3ea2285-50fc-4147-846c-f60994057b1e)




</details>
